src/widget/wstarrating.cpp

```diff
@@ -118,32 +82,24 @@ void WStarRating::mouseMoveEvent(QMouseEvent *event) {
 }
 
 void WStarRating::slotStarsUp(double v) {
-    if (!m_pCurrentTrack) {
-        return;
-    }
     if (v > 0 && m_starRating.starCount() < m_starRating.maxStarCount()) {
         int star = m_starRating.starCount() + 1;
-        m_starRating.setStarCount(star);
-        update();
-        m_pCurrentTrack->setRating(star);
+        emit ratingChangeRequest(star);
     }
 }
 
 void WStarRating::slotStarsDown(double v) {
-    if (!m_pCurrentTrack) {
-        return;
-    }
     if (v > 0 && m_starRating.starCount() > 0) {
         int star = m_starRating.starCount() - 1;
-        m_starRating.setStarCount(star);
-        update();
-        m_pCurrentTrack->setRating(star);
+        emit ratingChangeRequest(star);
```

Comment:
The signal should only be emitted if the value has actually changed to prevent cycles.

Reply:
Sure. This just handles the `group,stars_up` CO and changes the current value. The cycle prevention is in [`Track::setRating`](https://github.com/mixxxdj/mixxx/blob/2.4/src/track/track.cpp#L1363-L1368).

However, I changed this method to work with `m_currRating`, i.e. the stored value delivered by `Track::ratingUpdated`, not the visual state (which avoids confusion in the unlikely case a user hovers the rating **and** uses the CO).